ext/enchant/enchant.stub.php

```diff
@@ -2,94 +2,129 @@
 
 /** @generate-function-entries */
 
-/** @return resource|false */
-function enchant_broker_init() {}
+final class EnchantBroker
+{
+	/** @alias enchant_broker_init */
+	public function __construct() {}
 
-/** @param resource $broker */
-function enchant_broker_free($broker): bool {}
+	/** @alias enchant_broker_get_error */
+	public function getError(): string|false {}
+
+	/** @alias enchant_broker_list_dicts */
+	public function listDicts(): ?array {}
+
+	/** @alias enchant_broker_request_dict */
+	public function requestDict(string $tag): EnchantDict|false {}
+
+	/** @alias enchant_broker_request_pwl_dict */
+	public function requestPWL(string $filename): EnchantDict|false {}
+
+	/** @alias enchant_broker_dict_exists */
+	public function isDict(string $tag): bool {}
+
+	/** @alias enchant_broker_set_ordering */
+	public function setOrdering(string $tag, string $ordering): bool {}
+
+	/** @alias enchant_broker_describe */
+	public function describe(): ?array {}
+}
+
+final class EnchantDict
+{
+	public function __construct(EnchantBroker $broker, string $tag) {}
```

Comment:
I'm wondering if it really makes sense to have this constructor. It seems like there is not much benefit to allowing `new EnchantDict($broker, $tag)` instead of `$broker->requestDict($tag)`, just two ways to write the same thing.

Reply:
`requestDict` will return `false` in case of failure, which the `_construct` will raise an exception, which could seems better for OO dev.

BTW, I was thinking to have a way to construct PWL, perhaps

`public function __construct(EnchantBroker $broker, string $tagOrPath, bool $is_pwl);`

Reply:
Thinking a bit more:

* procedural way, use `enchant_broker_request_dict `and `enchant_broker_request_pwl_dict`
* OO way uses `EnchantDict::_construct()`

So `Enchant::requestDict()` and `Enchant::requestPWL() ` looks like some mixed of both way.... btw, implementation is no cost (simple alias). So I prefer to keep this __construct.

Reply:
> `requestDict` will return `false` in case of failure, while the `_construct` will raise an exception, which could seems better for OO dev.

If this is the concern, wouldn't it be better to make requestDict() throw in the OO API?

Reply:
> wouldn't it be better to make requestDict() throw in the OO API?

Indeed, done

Reply:
This is unresolved. @remicollet Please remove the EnchantDict constructor.

Reply:
But why ?
I really think having a class without a constructor is terrible idea.
User writing OO script will of course use the $dict = new EnchantDict(...)
ANd I really don't see any issue having multiple (3) ways to write it